colossalai/inference/engine/policies/chatglm2.py

```
@@ -1,29 +1,48 @@
+from functools import partial
 from typing import List
 
+import torch
 import torch.nn as nn
 
 from colossalai.shardformer.modeling.chatglm2_6b.modeling_chatglm import (
     ChatGLMForConditionalGeneration,
     ChatGLMModel,
     GLMBlock,
     GLMTransformer,
+    RMSNorm,
     SelfAttention,
 )
 
 # import colossalai
-from colossalai.shardformer.policies.chatglm2 import ChatGLMModelPolicy
+from colossalai.shardformer.policies.chatglm2 import ChatGLMForConditionalGenerationPolicy
 
 from ..modeling._utils import init_to_get_rotary
 from ..modeling.chatglm2 import ChatGLM2InferenceForwards
 
 try:
+    from colossalai.kernel.triton import rmsnorm_forward
+
     HAS_TRITON_RMSNORM = True
 except:
-    print("you should install triton from https://github.com/openai/triton")
+    print("Did not find rms-norm triton kernel")
+    print(
+        "You can use the following command to install: pip install git+https://github.com/ModelTC/lightllm.git@ece7b43f8a6dfa74027adc77c2c176cff28c76c8"
+    )
     HAS_TRITON_RMSNORM = False
 
 
-class ChatGLM2InferPolicy(ChatGLMModelPolicy):
+def get_triton_rmsnorm_forward():
+    if HAS_TRITON_RMSNORM:
+
+        def _triton_rmsnorm_forward(self: RMSNorm, hidden_states: torch.Tensor):
+            return rmsnorm_forward(hidden_states, self.weight.data, self.variance_epsilon)
+
+        return _triton_rmsnorm_forward
+    else:
+        return None
```

This should raise an exception

---

fixed